pkg/k8s/factory_functions.go

```diff
@@ -419,6 +419,10 @@ func equalV1Services(o1, o2 interface{}) bool {
 		return false
 	}
 
+	if !reflect.DeepEqual(svc1.GetAnnotations(), svc2.GetAnnotations()) {
```

Comment:
Please write a comment specifying why we use the annotations. also you have this available
```
comparator.MapStringEquals(svc1.GetAnnotations(), svc2.GetAnnotations())
```